homeassistant/components/tplink/config_flow.py

```diff
@@ -270,14 +272,37 @@ async def async_step_user(
         if user_input is not None:
             if not (host := user_input[CONF_HOST]):
                 return await self.async_step_pick_device()
-            self._async_abort_entries_match({CONF_HOST: host})
+            if "[" in host:
+                _, _, bracketed = host.partition("[")
+                host, _, port_str = bracketed.partition("]")
+                _, _, port_str = port_str.partition(":")
+            else:
+                host, _, port_str = host.partition(":")
+            match_dict = {CONF_HOST: host}
+            if port_str:
+                try:
+                    port = int(port_str)
+                    self.port = port
+                    match_dict[CONF_PORT] = port
+                except ValueError:
+                    port = None
+            else:
+                port = None
```

Comment:
Maybe extract this into its own function to keep the main flow cleaner?

Reply:
Agreed, that's done